src/pages/index.js

```diff
@@ -57,36 +89,134 @@ function openEditProfileForm() {
   profileEditPopup.open();
 }
 
+function openAvatarEditForm() {
+  formValidators["avatar-form"].resetValidation();
+  avatarEditPopup.open();
+}
+
 function openAddNewCardModal() {
   formValidators["card-form"].resetValidation();
   addCardPopup.open();
 }
 
 function handleProfileFormSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  profileEditPopup.close();
+  profileEditPopup.renderLoading(true);
+  api
+    .editProfileInfo(inputValues)
+    .then(() => {
+      userInfo.setUserInfo(inputValues);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+    });
 }
 
-function handleNewCardSubmit(inputValues) {
-  const { name, link } = inputValues;
-  const newCardEl = createCard({ name, link });
-  section.addItem(newCardEl);
-  addCardPopup.close();
+function handleNewAvatarSubmit(inputvalues) {
+  avatarEditPopup.renderLoading(true);
+  api
+    .updateProfilePicture(inputvalues)
+    .then(({ avatar }) => {
+      userInfo.setUserAvatar(avatar);
+      avatarEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarEditPopup.renderLoading(false);
+    });
+}
+
+function handleNewCardSubmit({ name, link }) {
+  addCardPopup.renderLoading(true);
+  api
+    .addNewCard({ name, link })
+    .then((data) => {
+      const newCardEl = createCard(data);
+      section.addItem(newCardEl);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
 }
 
 function createCard(data) {
-  const cardElement = new Card(data, "#card-template", handleCardClick);
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    handleCardClick,
+    handleDeleteBtnClick,
+    handleCardLike
+  );
   return cardElement.generateCard();
 }
 
 function handleCardClick(data) {
   cardPreviewPopup.open(data);
 }
 
+function handleCardLike(cardId) {
+  if (this.isLiked) {
+    api
+      .removeCardLike(cardId)
+      .then((card) => {
+        this.toggleLikes(card.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  } else {
+    api
+      .addCardLike(cardId)
+      .then((card) => {
+        this.toggleLikes(card.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+}
+
+function handleDeleteBtnClick(cardId) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: It's also needed to notify user about loading by changing submit button text